src/instructlab/sdg/pipeline.py

```diff
@@ -65,6 +66,7 @@ class PipelineContext:  # pylint: disable=too-many-instance-attributes
     dataset_num_procs: Optional[int] = DEFAULT_DATASET_NUM_PROCS
     checkpoint_dir: Optional[str] = None
     save_freq: Optional[int] = 1
+    max_num_tokens: Optional[int] = 4096
```

Comment:
We could also change this 4096 to `llmblock.DEFAULT_MAX_NUM_TOKENS` to reduce the hardcoded 4096 a bit more.